code/modules/awaymissions/corpse.dm

```diff
@@ -4,295 +4,530 @@
-			if(J.title == corpseidaccess)
-				jobdatum = J
-				break
-		if(corpseidicon)
-			W.icon_state = corpseidicon
-		if(corpseidaccess)
-			if(jobdatum)
-				W.access = jobdatum.get_access()
-			else
+		if(H.set_species(mob_species))
+			H.regenerate_icons()
+	if(husk)
+		H.ChangeToHusk()
+	else //Because for some reason I can't track down, things are getting turned into husks even if husk = false. It's in some damage proc somewhere.
+		H.mutations.Remove(HUSK)
+	H.underwear = "Nude"
+	H.undershirt = "Nude"
+	H.socks = "Nude"
+	var/obj/item/organ/external/head/D = H.get_organ("head")
+	if(istype(D))
+		if(hair_style)
+			D.h_style = hair_style
+		else
+			D.h_style = random_hair_style(gender)
+		if(facial_hair_style)
+			D.f_style = facial_hair_style
+		else
+			D.f_style = random_facial_hair_style(gender)
+	if(skin_tone)
+		H.s_tone = skin_tone
+	else
+		H.s_tone = random_skin_tone()
+	H.update_hair()
+	H.update_body()
+	if(outfit)
+		var/static/list/slots = list("uniform", "r_hand", "l_hand", "suit", "shoes", "gloves", "ears", "glasses", "mask", "head", "belt", "r_pocket", "l_pocket", "back", "id", "neck", "backpack_contents", "suit_store")
+		for(var/slot in slots)
+			var/T = vars[slot]
+			if(!isnum(T))
+				outfit.vars[slot] = T
+		H.equipOutfit(outfit, TRUE)
+		var/list/del_types = list(/obj/item/pda, /obj/item/radio/headset)
+		for(var/del_type in del_types)
+			var/obj/item/I = locate(del_type) in H
+			qdel(I)
+
+		if(disable_pda)
+			// We don't want corpse PDAs to show up in the messenger list.
+			var/obj/item/pda/PDA = locate(/obj/item/pda) in H
+			if(PDA)
+				var/datum/data/pda/app/messenger/M = PDA.find_program(/datum/data/pda/app/messenger)
+				M.toff = 1
+		if(disable_sensors)
+			// Using crew monitors to find corpses while creative makes finding certain ruins too easy.
+			var/obj/item/clothing/under/C = H.w_uniform
+			if(istype(C))
+				C.sensor_mode = SUIT_SENSOR_OFF
+
+	var/obj/item/card/id/W = H.wear_id
+	if(W)
+		if(id_access)
+			for(var/jobtype in typesof(/datum/job))
+				var/datum/job/J = new jobtype
+				if(J.title == id_access)
+					W.access = J.get_access()
+					break
+		if(id_access_list)
+			if(!islist(W.access))
 				W.access = list()
-		if(corpseidjob)
-			W.assignment = corpseidjob
-		W.registered_name = M.real_name
-		M.equip_to_slot_or_del(W, slot_wear_id)
-	if(coffin == 1)
-		var/obj/structure/closet/coffin/sarcophagus/sarc = locate(/obj/structure/closet/coffin/sarcophagus) in loc
-		if(sarc) M.loc = sarc
-	qdel(src)
-
-// I'll work on making a list of corpses people request for maps, or that I think will be commonly used. Syndicate operatives for example.
-/obj/effect/landmark/corpse/damaged
+			W.access |= id_access_list
+		if(id_job)
+			W.assignment = id_job
+		W.registered_name = H.real_name
+		W.update_label()
+
+//Instant version - use when spawning corpses during runtime
+/obj/effect/mob_spawn/human/corpse
+	roundstart = FALSE
+	instant = TRUE
+
+/obj/effect/mob_spawn/human/corpse/damaged

```

Comment:
If you want to implement the gib corpse spawners... Hmm, `create` itself doesn't offer a great interface for overriding. What if you added a `post_create` proc that you could override for the gib corpse subtype?